src/webapp/gradebookServer.js

```diff
@@ -190,12 +190,24 @@ app.get('/seasons', function(request, response) {
       passwordText, request.query.host, request.query.port);
 
    //Get the params from the url
-   var instructorID = request.query.instructorid;
+   var userID = request.query.userid;
+   var userRole = request.query.userRole;
    var year = request.query.year;
-
-   //Set the query text
-   var queryText = 'SELECT SeasonOrder, SeasonName FROM gradebook.getInstructorSeasons($1, $2);';
-   var queryParams = [instructorID, year];
+   var queryText;
+   var queryParams;
+
+   //Set the query
+   var queryParams = [userID, year];
+
+   if(userRole == 'alpha_GB_Instructor') {
+      queryText = 'SELECT Order, Name FROM getInstructorSeasons($1, $2);';
+   }
+   else if(userRole == 'alpha_GB_Student') {
+      queryText = 'SELECT Order, Name FROM getStudentSeasons($1, $2);';
+   }
+   else {
+      queryText = 'SELECT DISTINCT season FROM term WHERE Year = ($2);';
```

Comment:
```suggestion
      queryText = 'SELECT S."Order", S.Name FROM term T JOIN season S ON ' +
                  'T.season = S."Order" WHERE T.Year = $2;';
```
This change corrects a few issues with the previous query:

1. There's not really a need to use `DISTINCT` here, since only one of each season can be associated with a year.
2. The result schema (columns) of the earlier query did not match that of the other two queries in the conditional statement, which would cause issues when parsing the result.
3. No need for parentheses around `$2` (it is not being used as a parameter to an SQL function)